lms/templates/discussion/_user_profile.html

```diff
@@ -3,8 +3,7 @@
 <div class="user-profile">
   <div class="sidebar-username"><a class="learner-profile-link" href="${learner_profile_page_url}">${django_user.username | h}</a></div>
   <div class="sidebar-user-roles">
-    <% role_names = django_user.roles.order_by("name").values_list("name", flat=True).distinct() %>
```

Comment:
Perfectly fine. I can help you understand. 

Actually, I moved the roles logic in the view and passed from there. Check `user_roles` context variable.